### PR TITLE
feat: add postgresql/prefer-text-over-varchar rule

### PR DESCRIPTION
**Pull request by baseballyama**
## Summary

New rule `postgresql/prefer-text-over-varchar` that warns on `varchar(n)` columns. Enabled at `warn` in `configs.recommended`.

## Motivation

PostgreSQL stores `text` and `varchar(n)` the same way internally. The length is a constraint that you cannot relax without a full table rewrite, which makes it a poor place to encode "we want at most N chars" — that belongs in a `CHECK` constraint that can be edited via `ALTER TABLE ... DROP CONSTRAINT` + a new one. `VARCHAR` without a length is unaffected.

## Stacked on #60

Branches off `chore/oss-scaffolding` (#60). Merge #60 first.

## Changes

- New rule `postgresql/prefer-text-over-varchar` (only fires when `varchar` has a typmod).
- Wired into `configs.recommended` at `warn`.
- README rule docs with the `CHECK (length(...) <= N)` migration suggestion.
- Unit test + fixtures.
- Changeset (`minor`).

## Testing

`pnpm check:all` and `pnpm test` pass locally.

## Breaking changes

New default-on warning in `configs.recommended` (pre-1.0). Flagged in the changeset.

## Checklist

- [x] I have read CLAUDE.md and followed the project's conventions.
- [x] I have added or updated tests for the change.
- [x] I have added or updated documentation where user-visible behavior changed.
- [x] If this is a breaking change, I have added a changeset and flagged it above.
- [x] I have re-read my own diff and removed dead code, debug prints, and stale comments.
- [x] If I used an LLM to draft this PR, I have verified each change myself, this PR represents real work that warrants a maintainer's review, and I am willing to defend each line in review.

### CODE REVIEW

File: src/rules/prefer-text-over-varchar.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+const getTypeName = (typeName: any): string | undefined => {
+  if (!typeName || typeof typeName !== "object") return undefined;
+  const seg0 = typeName["0"];
+  const seg1 = typeName["1"];
+  const v0 = seg0?.sval;
+  const v1 = seg1?.sval;
+  if (v0 === "pg_catalog" && typeof v1 === "string") return v1;
+  if (typeof v0 === "string") return v0;
+  return undefined;
```
[the remark this answers is not in the document]
Comment (baseballyama) [reply src/rules/prefer-text-over-varchar.ts]: Fixed: `getTypeName` now prefers segment-1 when present, so `public.varchar(255)` correctly resolves to `varchar` instead of returning the schema. Same fix applied across the other type-checking rules (`prefer-jsonb-over-json`, `prefer-identity-over-serial`, `prefer-timestamptz`, `no-money-type`, `no-char-type`).

File: CLAUDE.md [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+### Tests
+
+Two layers:
+
+1. **`tests/<rule>.test.ts`** — RuleTester-style cases with inline SQL strings
+   for the canonical happy path and a few edge cases.
+2. **`tests/fixtures/<rule>/`** — larger SQL fixtures snapshot-tested against
+   the rule's output. Regenerate with `pnpm update-fixtures` when the change
+   is intentional, and review the diff carefully before committing.
```
[the remark this answers is not in the document]
Comment (baseballyama) [reply CLAUDE.md:190]: Good catch — that mismatch lives in `CLAUDE.md` (which has merged via #60), not in this PR's diff after the rebase. Worth a follow-up against `main` to update the testing-section example.

File: pnpm-workspace.yaml [hunk 3]
```diff
@@ -1,2 +1,9 @@
 onlyBuiltDependencies:
+  - "@launchql/protobufjs"
+  - esbuild
   - libpg-query
+# pnpm 11 makes strictDepBuilds true by default; we opt out so CI does not
+# fail when a transitive dependency adds a new build script. The packages we
+# actually need built are listed above; anything else is intentionally not
+# run.
```
[the remark this answers is not in the document]
Comment (baseballyama) [reply pnpm-workspace.yaml:8]: After the rebase onto `origin/main` (#60 merged), `pnpm-workspace.yaml` is no longer in this PR's diff.

File: README.md [hunk 4]
```diff
@@ -84,6 +84,31 @@ SELECT * FROM WHERE id = 1;
 SELECT * FROM users WHERE id = 1;
 ```
 
+### `postgresql/prefer-text-over-varchar`
+
+Warns on `varchar(n)` columns. PostgreSQL stores `text` and `varchar(n)` the same way internally; the length is enforced by a per-table constraint that you cannot relax without rewriting the table. Use `text` and add a `CHECK (length(col) <= N)` constraint when you actually need a cap.
+
+**Type**: Suggestion
+**Recommended**: ⚠️ Warn
+**Fixable**: ❌ No
```
[the remark this answers is not in the document]
Comment (baseballyama) [reply README.md]: Fixed: added the trailing two spaces so the metadata renders as separate lines.
